src/gt4py/next/iterator/transforms/inline_dynamic_shifts.py

```diff
@@ -0,0 +1,76 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import dataclasses
+from typing import Optional
+
+import gt4py.next.iterator.ir_utils.common_pattern_matcher as cpm
+from gt4py import eve
+from gt4py.eve import utils as eve_utils
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.transforms import fuse_as_fieldop, inline_lambdas, trace_shifts
+from gt4py.next.iterator.transforms.symbol_ref_utils import collect_symbol_refs
+
+
+def _dynamic_shift_args(node: itir.Expr) -> None | list[bool]:
+    if not cpm.is_applied_as_fieldop(node):
+        return None
+    params_shifts = trace_shifts.trace_stencil(
+        node.fun.args[0],  # type: ignore[attr-defined]  # ensured by is_applied_as_fieldop
+        num_args=len(node.args),
+        save_to_annex=True,
+    )
+    dynamic_shifts = []
+    for param_shifts in params_shifts:
+        has_dynamic_shift = False
+        for shifts in param_shifts:
+            for _, offset in zip(shifts[::2], shifts[1::2], strict=True):
```

Comment:
either 
```suggestion
            for offset in shifts[1::2]:
```
or use a function if you insist that does proper error checking.